pkgs/servers/sql/postgresql/generic.nix

```diff
@@ -641,76 +641,84 @@ let
-              lib
-              makeBinaryWrapper
-              postgresql
-              ;
-          } (_: installedExtensions ++ [ postgresql.jit ]);
-          withoutJIT = postgresqlWithPackages {
-            inherit
-              buildEnv
-              lib
-              makeBinaryWrapper
-              postgresql
+      finalPackage =
+        (buildEnv {
+          name = "${postgresql.pname}-and-plugins-${postgresql.version}";
+          paths = installedExtensions ++ [
+            # consider keeping in-sync with `postBuild` below
+            postgresql
+            postgresql.man # in case user installs this into environment
+          ];
+
+          pathsToLink = [
+            "/"
+            "/bin"
+            "/share/postgresql/extension"
+            # Unbreaks Omnigres' build system
+            "/share/postgresql/timezonesets"
+            "/share/postgresql/tsearch_data"
+          ];
+
+          nativeBuildInputs = [ makeBinaryWrapper ];
+          postBuild =
+            let
+              args = lib.concatMap (ext: ext.wrapperArgs or [ ]) installedExtensions;
+            in
+            ''
+              wrapProgram "$out/bin/postgres" ${lib.concatStringsSep " " args}
+
+              mkdir -p "$dev/nix-support"
+              substitute "${lib.getDev postgresql}/nix-support/pg_config.env" "$dev/nix-support/pg_config.env" \
+                --replace-fail "${postgresql}" "$out" \
+                --replace-fail "${postgresql.man}" "$out"
+            '';
+
+          passthru = {
+            inherit installedExtensions;
+            inherit (postgresql)
+              pkgs
+              psqlSchema
+              version
               ;
-          } (_: lib.remove postgresql.jit installedExtensions);
 
-          withPackages =
-            f':
-            postgresqlWithPackages {
+            pg_config = postgresql.pg_config.override { inherit finalPackage; };
+
+            withJIT = postgresqlWithPackages {
               inherit
                 buildEnv
                 lib
                 makeBinaryWrapper
                 postgresql
                 ;
-            } (ps: installedExtensions ++ f' ps);
-        };
-      };
+            } (_: installedExtensions ++ [ postgresql.jit ]);
+            withoutJIT = postgresqlWithPackages {
+              inherit
+                buildEnv
+                lib
+                makeBinaryWrapper
+                postgresql
+                ;
+            } (_: lib.remove postgresql.jit installedExtensions);
+
+            withPackages =
+              f':
+              postgresqlWithPackages {
+                inherit
+                  buildEnv
+                  lib
+                  makeBinaryWrapper
+                  postgresql
+                  ;
+              } (ps: installedExtensions ++ f' ps);
+          };
+        }).overrideAttrs
+          {
+            # buildEnv doesn't support passing `outputs`, so going via overrideAttrs.
```

Comment:
Not a big fan of this, but OK.

Reply:
Hmm, I haven't tested yet, but are there ways this could mess up splicing?

Like, the fix is good enouh and I done pretty much the same, but I do wonder if it'd be OK to just accept `outputs` in `buildEnv`. Like, in the end it's another derivation (that can be modified with overrideAttrs -- as done right now), so I'd argue this is OK even though it's technically a larger API surface :thinking: 

Thoughts?